packages/libp2p/src/connection-manager/index.ts

```diff
@@ -575,7 +601,7 @@ export class DefaultConnectionManager implements ConnectionManager, Startable {
   async acceptIncomingConnection (maConn: MultiaddrConnection): Promise<boolean> {
     // check deny list
     const denyConnection = this.deny.some(ma => {
-      return maConn.remoteAddr.toString().startsWith(ma.toString())
+      return ma.contains(maConn.remoteAddr.nodeAddress().address)
```

Comment:
Prefer `toOptions` because `nodeAddress` will throw if the protocol isn't `tcp` or `udp`:

```suggestion
      return ma.contains(maConn.remoteAddr.toOptions().host)
```